### ISSUE TITLE
Defining `null` defaults for variables makes checking more difficult

### ISSUE THREAD

**Issue by eengstrom** [maxhoesel/ansible-collection-smallstep#20]
I sort of understand the reason to make sure that every variable has a default (`defaults/main.yml`), but in the case of defining variables with a `null` default, it makes checking for appropriate values more difficult.  As it stands now, if you don't define a value for either `stepca_root_password` or `stepca_intermediate_password`, the check for validity will fail, not with your defined failure message, but with a true error (`NoneType has no len()`).

I would suggest commenting out the definitions in `defaults/main.yml` and adding revising the assert clauses, as in:

```
- name: Check that passwords are set
  assert:
    that:
      - stepca_root_password is defined
      - stepca_root_password | length > 0
      - stepca_intermediate_password is defined
      - stepca_intermediate_password | length > 0
    fail_msg: "Please make sure that both stepca_root_password and stepca_intermediate_password are set!"
```

This actually makes it easier to conditionalize use of the role in my standard usage as well, with a `pre_task` such as:

```
  pre_tasks:
    # Allow this playbook to be run on ALL hosts,
    # BUT, skip everything unless required vars are defined for this host.
    - name: step-ca | End pla when insufficient variables defined
      meta: end_host
      when: stepca_root_password is defined
```

**Comment by eengstrom** [comment]
BTW - didn't want to presume, but if you want PRs, just say so - happy to contribute.

**Comment by maxhoesel** [comment]
Thanks for the suggestion! I'm planning to rewrite the ca role soon anyway (see [here](https://github.com/maxhoesel/ansible-collection-smallstep/issues/21), so this seems like a good improvement and logical inclusion. 

As for PRs, I'm more than happy to accept them, but as I'm currently still in the process of working out the fundamentals for this collection, I'm not sure if it'd be such a good idea right now - there's good chance that i'll rip out parts of the collection and rework them soon, and I don't want to waste anyone else's time until things are at least somewhat stable. Still a good way to go until that's the case, hah

**Comment by maxhoesel** [comment]
This should now be addressed in the 0.3_rework branch using the new step_cli/ca roles. I'll leave this issue open until the branch is merged, just in case any other issues pop up

**Comment by maxhoesel** [comment]
I just released 0.3.0, which should address this issue
